spectroscopy/code_src/Euclid_functions.py

```diff
@@ -0,0 +1,123 @@
+import astropy.units as u
+import numpy as np
+import pandas as pd
+from astropy.coordinates import SkyCoord
+from astropy.table import QTable
+from astropy.io import fits
+from astroquery.ipac.irsa import Irsa
+
+from data_structures_spec import MultiIndexDFObject
+import warnings
+from astropy.utils.exceptions import AstropyWarning
+
+warnings.simplefilter('ignore', category=AstropyWarning)
+
+
+def get_coord_from_objectid(object_id, table_mer="euclid_q1_mer_catalogue"):
+    adql_query = f"""
+    SELECT ra, dec FROM {table_mer}
+    WHERE object_id = {object_id}
+    """
+    try:
+        result = Irsa.query_tap(adql_query).to_table()
+        if len(result) == 0:
+            print(f"No match found for object_id {object_id}")
+            return None
+        ra, dec = result[0]["ra"], result[0]["dec"]
+        return SkyCoord(ra=ra * u.deg, dec=dec * u.deg)
+    except Exception as e:
```

Comment:
I'll let @troyraen or @bsipocz  comment, but I know they don't like these very broad "except Exceptions" and would rather have a specific exception caught if it is an issue.   Here and below..

Reply:
yes, please be specific with exceptions. Also, I wonder if we even need a try/except here as you already did a check for result length --> propagate the issue to the user as it's a real one rather than quietly swallow it.